src/cfnlint/transform.py

```diff
@@ -0,0 +1,161 @@
+        S3 Uri. This is to support running the SAM Translator with valid values for these fields. If this in not done,
+        the template is invalid in the eyes of SAM Translator (the translator does not support local paths)
+        """
+
+        all_resources = self._template.get('Resources', {})
+
+        for _, resource in all_resources.items():
+
+            resource_type = resource.get('Type')
+            resource_dict = resource.get('Properties')
+
+            if resource_type == 'AWS::Serverless::Function':
+
+                Transform._update_to_s3_uri('CodeUri', resource_dict)
+
+            if resource_type == 'AWS::Serverless::Api':
+                if 'DefinitionBody' not in resource_dict:
+                    Transform._update_to_s3_uri('DefinitionUri', resource_dict)
+
+    def transform_template(self):
+        """
+        Transform the Template using the Serverless Application Model.
+        """
+        matches = []
+
+        # Boto's mechanism is used to fetch the region. Since we know which we use
+        # to lint, make sure Boto loads that one.
+        # https://github.com/awslabs/serverless-application-model/blob/master/samtranslator/translator/arn_generator.py#L39
+        # https://github.com/awslabs/serverless-application-model/blob/master/tests/translator/test_translator.py#L120
+        old_region = os.environ.get('AWS_DEFAULT_REGION', '')
+        os.environ['AWS_DEFAULT_REGION'] = self._region
+
+        try:
+            sam_translator = Translator(managed_policy_map=self._managed_policy_map,
+                                        sam_parser=self._sam_parser)
+
+            self._replace_local_codeuri()
+
+            # In the Paser class, within the SAM Translator, they log a warning for when the template
+            # does not match the schema. The logger they use is the root logger instead of one scoped to
+            # their module. Currently this does not cause templates to fail, so we will suppress this
+            # by patching the logging.warning method that is used in that class.
+            class WarningSuppressLogger(object):
+                """ Patch the Logger in SAM """
+
+                def __init__(self, obj_to_patch):
+                    self.obj_to_patch = obj_to_patch
+
+                def __enter__(self):
+                    self.obj_to_patch.warning = self.warning
+
+                def __exit__(self, exc_type, exc_val, exc_tb):
+                    self.obj_to_patch.warning = self.obj_to_patch.warning
+
+                def warning(self, message):
+                    """ Ignore warnings from SAM """
+                    pass
+
+            with WarningSuppressLogger(parser.logging):
+                sam_translator.translate(sam_template=self._template, parameter_values={})
+        except InvalidDocumentException as e:
+            for cause in e.causes:
+                matches.append(cfnlint.Match(
```

Comment:
There may not be much we can do here but it would be nice to get into line and column numbers.  Maybe we can provide some help back to them on this part :)

Reply:
Removed this and covered it in TOX now.

Reply:
Line numbers would be cool. I can see this being a huge improvement for customers using SAM. Would you mind creating a [GH Issue](https://github.com/awslabs/serverless-application-model/issues) for it?

Reply:
I'll take care of that 👍